### PR TITLE
Node to uint32_t

### PR DESCRIPTION
**Pull request by etwest**

### CODE REVIEW

**Comment by etwest** [comment]
Nothing needs to be changed on the streaming side of things so this should be a simple process.

File: include/buffer_tree.h [hunk 1]
```diff
@@ -169,7 +169,7 @@ class BufferTree {
    * we would like to be accesible to all the bufferControlBlocks
    */
   static uint32_t page_size;
-  static const uint32_t serial_update_size = sizeof(Node) + sizeof(Node);
+  static const uint serial_update_size = sizeof(Node_ID) + sizeof(Node_ID);
```
Comment (Victor-C-Zhang) [inline include/buffer_tree.h:172]: don't use uint. it's not portable.

**Comment by Victor-C-Zhang** [comment]
Whtat's the point of this change?

File: include/buffer_tree.h (hunk 1 above)
Comment (etwest) [reply include/buffer_tree.h:172]: uints are sneaky buggers. I'll get rid of it

**Comment by etwest** [comment]
> Whtat's the point of this change?

The point is to halve the memory requirements of the buffering systems both Gutter tree and in-RAM. We'll also have to change the work_queue.cpp file to reflect this change. I'll submit a pull request for that once this is in.

**Comment by Victor-C-Zhang** [comment]
Perhaps make it clearer that it's a node ID type. Maybe something like `node_id_t`?

**Review by Victor-C-Zhang — APPROVED** [review]
Take a look at my comment. You don't need to make the changes now, but we should deal with that at some point.

**Review by Victor-C-Zhang — APPROVED** [review]
LGTM

**Pull request merged**